pkg/stores/tournament/cache.go

```diff
@@ -27,6 +27,7 @@ type backingStore interface {
 	RemoveRegistrants(ctx context.Context, tid string, userIDs []string, division string) error
 	RemoveRegistrantsForTournament(ctx context.Context, tid string) error
 	ActiveTournamentsFor(ctx context.Context, userID string) ([][2]string, error)
+	GetRecentAndUpcomingTournaments(ctx context.Context) ([]*entity.Tournament, error)
```

Comment:
This is a super simple API that's basically hard-coded for the "show some tournaments on the homepage" use-case.

I'm pretty open to other suggestions here which may be more extensible in the future